### PR TITLE
Revert "Revert "Pyarrow Segfault Regression Test (#7568) (#7805)

### PR DESCRIPTION
**Pull request by ijrsvt**
## Related issue number

#7805 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: doc/requirements-doc.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@ opencv-python-headless
 pandas
 pickle5
 pillow
+pyarrow
```
Comment (chaokunyang) [inline doc/requirements-doc.txt]: Can you add `pyarrow` in `python/setup.py` and `ci/travis/install-dependencies.sh`. And I think we shopuld enforece pyarrow version to ge greater than 0.15

Comment (ijrsvt) [reply doc/requirements-doc.txt]: Thanks! This is definitely not the right location.

**Comment by rkooo567** [comment]
@ijrsvt Is this a critical PR? Do you think it should be merged before the next relesase?

**Comment by simon-mo** [comment]
@chaokunyang can I ask what's the status on this PR? Is it working? can you fix merge some merge conflict? I'm happy to merge this PR given test passes.

**Comment by ijrsvt** [comment]
Stale & not causing problems for the time being

**Pull request closed without merging**